hack/deployer/runner/client.go

```diff
@@ -77,13 +83,28 @@ func clientImageName(driverID, clientVersion, dockerfileName string) (string, er
 	return fmt.Sprintf("%s-%s:%s-%.8x", clientBaseImageName, driverID, clientVersion, h.Sum(nil)), nil
 }
 
-func dockerLogin() error {
-	registryEnv := ".registry.env"
-	if _, err := os.Stat(registryEnv); os.IsNotExist(err) {
-		// not attempting login when registry env file does not exist (typically outside of CI)
-		log.Printf("Not attempting Docker login as .registry.env is not present in the filesystem.")
-		return nil
+func dockerLogin(client vault.Client) error {
+	// skip docker login in dev if registry exists in docker auth config file
+	if os.Getenv("CI") != "true" {
```

Comment:
Took me a while to figure out that this is set by buildkite. Is this code path no longer used for any Jenkins builds?

Reply:
This code path is used for Buildkite and Jenkins. But setting `CI=true` in the build environment is a standard practice done by almost all known CI systems.

Reply:
TIL!